x-pack/plugins/asset_manager/server/lib/implicit_collection/collectors/containers.ts

```diff
@@ -5,24 +5,27 @@
  * 2.0.
  */
 
+import { estypes } from '@elastic/elasticsearch';
 import { Asset } from '../../../../common/types_api';
 import { CollectorOptions, QUERY_MAX_SIZE } from '.';
 import { withSpan } from './helpers';
 
 export async function collectContainers({
   client,
   from,
+  to,
   transaction,
   sourceIndices,
+  afterKey,
 }: CollectorOptions) {
   const { metrics, logs, traces } = sourceIndices;
-  const dsl = {
+  const dsl: estypes.SearchRequest = {
     index: [traces, logs, metrics],
     size: QUERY_MAX_SIZE,
     collapse: {
       field: 'container.id',
     },
-    sort: [{ _score: 'desc' }, { '@timestamp': 'desc' }],
+    sort: [{ 'container.id': 'asc' }],
```

Comment:
_copied from slack discussion_
by sorting on hostname for pagination purposes it seems that the scoring is based on the sort rather than the query, which means the top collapsed document is not necessarily the most relevant given our filters. However when querying for an inner hit (with `collapse.inner_hits: { size: 1 }`) it seems we actually get the top scoring document for each group as additional payload but this comes with important performance hit. Another approach would be to filter out documents that don't have the fields we're hoping for